model/QueueBroker/NewSqlQueueBroker.php

```diff
@@ -0,0 +1,271 @@
+     *
+     * @inheritdoc
+     */
+    public function createQueue()
+    {
+        $persistence = $this->getPersistence();
+        /** @var AbstractSchemaManager $schemaManager */
+        $schemaManager = $persistence->getDriver()->getSchemaManager();
+
+        /** @var Schema $schema */
+        $schema = $schemaManager->createSchema();
+        $fromSchema = clone $schema;
+
+        if (in_array($this->getTableName(), $schemaManager->getTables())) {
+            $schema->dropTable($this->getTableName());
+        }
+        $queries = $persistence->getPlatForm()->getMigrateSchemaSql($fromSchema, $schema);
+
+        foreach ($queries as $query) {
+            $persistence->exec($query);
+        }
+
+        try {
+            $this->getSchemaProvider($schema)
+                ->setQueueName($this->getQueueName())
+                ->getSchema($schema, $this->getTableName());
+        } catch (Throwable $e) {
+            $this->logDebug('Schema of ' . $this->getTableName() . ' table already up to date.');
+        }
+
+        $queries = $persistence->getPlatForm()->getMigrateSchemaSql($fromSchema, $schema);
+
+        foreach ($queries as $query) {
+            $persistence->exec($query);
+        }
+
+        if ($queries) {
+            $this->logDebug('Queue ' . $this->getTableName() . ' created/updated in RDS.');
+        }
+    }
+
+    public function getSchemaProvider(Schema $schema): NewSqlSchema
+    {
+        return $this->getServiceLocator()->get(NewSqlSchema::class);
+    }
+
+    /**
+     * Insert a new task into the queue table.
+     *
+     * @param TaskInterface $task
+     * @return bool
+     */
+    public function push(TaskInterface $task)
+    {
+        return (bool)$this->getPersistence()->insert($this->getTableName(), [
+            'id' => $this->getUniquePrimaryKey(),
+            'message' => $this->serializeTask($task),
+            'created_at' => $this->getPersistence()->getPlatForm()->getNowExpression(),
+            'visible' => true,
+        ]);
+    }
+
+    /**
+     * Does the DBAL specific pop mechanism.
+     */
+    protected function doPop()
+    {
+        $this->getPersistence()->getPlatform()->beginTransaction();
+
+        $logContext = [
+            'Queue' => $this->getQueueNameWithPrefix()
+        ];
+
+        try {
+            $qb = $this->getQueryBuilder()
+                ->select('id, message')
+                ->from($this->getTableName())
+                ->andWhere('visible = :visible')
+                ->orderBy('created_at')
+                ->setMaxResults($this->getNumberOfTasksToReceive());
+
+            /**
+             * SELECT ... FOR UPDATE is used for locking
+             *
+             * @see https://dev.mysql.com/doc/refman/5.6/en/innodb-locking-reads.html
+             */
+            $sql = $qb->getSQL() . ' ' . $this->getPersistence()->getPlatForm()->getWriteLockSQL();
```

Comment:
Instead of comments, we might document the code with self-explainable methods:

```
private function getSQLWithLocking(QueryBuilder $queryBuilder): string
{
    return $queryBuilder->getSQL() . ' ' . $this->getPersistence()->getPlatForm()->getWriteLockSQL();
}
```